pilot/pkg/model/sidecar.go

```diff
@@ -0,0 +1,324 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"strings"
+
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+const (
+	wildcardNamespace = "*"
+	wildcardService   = Hostname("*")
+)
+
+// SidecarScope is a wrapper over the Sidecar resource with some
+// preprocessed data to determine the list of services, virtualServices,
+// and destinationRules that are accessible to a given sidecar.
+type SidecarScope struct {
+	// The crd itself. Can be nil if we are constructing the default
+	// sidecar scope
+	Config *Config
+
+	// set of egress listeners, and their associated services.
+	// A sidecar scope should have either ingress/egress listeners or both.
+	EgressListeners []*IstioEgressListenerWrapper
+
+	// // set of ingress listeners
+	// // A sidecar scope should have either ingress/egress listeners or both.
+	// IngressListeners []*IstioIngressListenerWrapper
+
+	// Union of services imported across all egress listeners for use by CDS code.
+	// Right now, we include all the ports in these services.
+	// TODO: Trim the ports in the services to only those referred to by the
+	// egress listeners.
+	allImportedServices []*Service
```

Comment:
Not sure what 'all' means here - importedServices should be enough. Do we have a case for an incomplete list (for on-demand ) ? Confused.

Reply:
See the comment. "Union of services imported *across all egress listeners*". I dont know what more comments I can add to clarify what this field represents.

Reply:
You mean the scope will combine 'egress' declarations from all matching Sidecar resources ? 
I assumed we match the most specific Sidecar, I don't think merging Sidecars is right.

Reply:
Dude read the api please. This is becoming very frustrating, trying to make you understand the code when you refuse to read the API specs. A sidecar has multiple egress listeners as per the API. and this is combining services across all listeners in the sidecar.